src/challenges/challengeOne/4.Form/Form.test.tsx

```diff
@@ -1,8 +1,68 @@
-// import React from "react";
-// import { render, screen } from "@testing-library/react";
-// import userEvent from "@testing-library/user-event";
-// import { Form } from "./Form";
-
-// describe("Form", () => {
-//   it("renders...", () => {});
-// });
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { Form, FormProps, FormField } from "./Form";
+
+describe("Form", () => {
+  const onSubmit = jest.fn();
+
+  const renderComponent = (additionalProps?: Partial<FormProps>) => {
+    const fields: FormField[] = [
+      {
+        label: "Label 1",
+        type: "text",
+        name: "field1",
+        placeholder: "Placeholder 1",
+        value: "",
+        id: "field1-id",
+      },
+      {
+        label: "Label 2",
+        type: "text",
+        name: "field2",
+        placeholder: "Placeholder 2",
+        value: "",
+        id: "field2-id",
+      },
+    ];
+
+    const defaultProps: FormProps = {
+      fields,
+      onSubmit,
+    };
+
+    return render(<Form {...defaultProps} {...additionalProps} />);
+  };
+
+  it("renders the correct number of fields", () => {
+    renderComponent();
+    expect(screen.getAllByRole("textbox")).toHaveLength(2);
+  });
+
+  it("updates the field value when typed into", async () => {
```

Comment:
I would have added tests for both inputs, because what if the second one was changed and broke, we wouldn't have a failing test to alert us.